### PR TITLE
InstructorResults: filter by section options missing "Not in a section" option #4057

### LINKED ISSUE #4057: InstructorResults: filter by section options missing "Not in a section" option
filterbysection

If there are no students that are not a in section, then the option for "Not in a section" does not appear in the option to filter responses by section. We should add that in if the instructors wants to view responses from/to Instructors, and to the "General" recipient, when there are too many responses

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #4057.

The question view might not display all the expected responses between students and instructors because of #4125, which should be fixed there. 
@yj-soh, assigning this to you since you migrated the filter panel to jstl

### CODE REVIEW

**Comment by yj-soh** [comment]
Should this still show when it's non-applicable? (i.e. when there aren't any user in that category)

**Comment by kanghj** [comment]
I think it should? After all, we show an option for every section even if there are sections without a response. When there are many responses in a section, the non-question views (RGQ, GRQ, RQG, GQR) also prepares a "Not in a section" panel for ajax loading even if it isn't applicable.

File: src/test/resources/pages/instructorFeedbackResultsPageOpen.html [hunk 1]
```diff
@@ -4912,4 +4920,4 @@
 </div>
 
 
-<div class="tooltip fade top in" style="top: {*}px; left: {*}px; display: block;"><div class="tooltip-arrow"></div><div class="tooltip-inner">View results by sections</div></div></body></html>
```
Comment (yj-soh) [inline src/test/resources/pages/instructorFeedbackResultsPageOpen.html]: What caused this?

Comment (kanghj) [inline src/test/resources/pages/instructorFeedbackResultsPageOpen.html]: Godmode. Whenever a tooltip appears when hovered over, this element is updated. It's ignored in our html tests.

**Pull request merged**